doc/api/next_api_changes/development/24893-KS.rst

```diff
@@ -0,0 +1,15 @@
+Maximum line length increased to 88 characters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The maximum line length for new contributions has been extended from 79 characters to
+88 characters.
+This change provides an extra 9 characters to allow code which is a single idea to fit
+on fewer lines (often a single line).
+Other line lengths considered included 115 and 99, but ultimately 88 characters was the
+consensus as it is relatively conservative while still providing reasonable benefit.
+The ability to view side-by-side git diffs was a key point in choosing the smaller
+limit.
```

Comment:
On a more serious note, I wonder about putting too much of our back and forth in the API note.  Maybe just say we chose 88, its consistent with black, but we may relax further if there is a good reason...